mirascope/base/vectorstores.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Optional, Union
+
+from pydantic import BaseModel
+
+from mirascope.base.chunkers import BaseChunker
+from mirascope.base.embedders import BaseEmbedder
+from mirascope.openai.embedders import OpenAIEmbedder
+
+from .types import BaseVectorStoreParams, Document
+
+
+class BaseVectorStore(BaseModel, ABC):
+    vectorstore_api_key: Optional[str] = None
+    index_name: ClassVar[Optional[str]] = None
+    chunker: ClassVar[BaseChunker] = BaseChunker()
```

Comment:
realizing this won't accept a chunker function instead of the class?

Reply:
hm they create a chunker class with params. and pass in the function to the chunker class, then pass the chunker class to the vectorstore. is this not what we want?

Reply:
I though we also wanted to allow for passing the function in directly and inferring the chunker by inspecting the function (i.e. set `chunking_params` to whatever the default value for `chunking_params` in the function is, and if no default is set use `BaseChunkingParams`?